### PR TITLE
Organize shimmed dependencies; add more shimmed dependencies

### PR DESCRIPTION
**Pull request by pl12133**
**Context:** We have Zimlet developers who would like to use `react-apollo` in their Zimlets, but right now they have to re-bundle it into the Zimlet rather than using the dependency from the main Application. This requires an accompanying PR to `zm-x-web` to re-shape the shims being passed as context.

Aliasing makes it totally seamless, the consumer just uses `import` and everything works like magic.

There are a bunch of good candidates for shimming:

 - `preact-i18n`
 - `preact-redux`
 - `wiretie`
 - `redux`

These changes are not backwards compatible.

### CODE REVIEW

File: src/shims/preact-router.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/** This file is a shim, aliased in for "preact-router" in the webpack config.
+ *  When components import 'preact-router', we want to give them back the copy
+ *  Zimbra passed down when it called the factory provided to zimlet().
+ */
+
+const { route, Link } = global.shims.preactRouter;
```
Comment (billneff79) [inline src/shims/preact-router.js]: does `export const { route, Link } = global.shims.preactRouter` not work?

File: src/index.js [hunk 2]
```diff
@@ -153,7 +153,9 @@ export function configure(env) {
 			],
 
 			alias: {
-				preact: path.resolve(__dirname, 'preact.js'),
+				preact: path.resolve(__dirname, 'shims/preact.js'),
+				'preact-router': path.resolve(__dirname, 'shims/preact-router.js'),
+				'react-apollo': path.resolve(__dirname, 'shims/react-apollo.js'),
```
Comment (billneff79) [inline src/index.js]: Other obvious one that jumps out at me is `preact-redux` that we might want to add.

File: src/index.js [hunk 3]
```diff
@@ -153,7 +153,9 @@ export function configure(env) {
...
 				react: 'preact',
```
Comment (billneff79) [inline src/index.js]: Similar to your note on Darby's PR, should this be `react: 'preact-compat'`

Comment (pl12133) [reply src/index.js]: Fixed, and that allows this PR to obsolete https://github.com/Zimbra/zimlet-cli/pull/5.

File: src/shims/preact-router.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/** This file is a shim, aliased in for "preact-router" in the webpack config.
+ *  When components import 'preact-router', we want to give them back the copy
+ *  Zimbra passed down when it called the factory provided to zimlet().
+ */
+
+export const { route, Link, Match } = global.shims.preactRouter;
```
Comment (billneff79) [inline src/shims/preact-router.js]: `Match` is actually an export of `preact-router/match` not of `preact-router`.  Might want a separate alias/shim for that

Comment (billneff79) [reply src/shims/preact-router.js]: - [ ] remove `Match` and create alias for `preact-router/match` if desired

**Comment by pl12133** [comment]
Everything in 3055a2b seems to work fine in the Notepad Zimlet but I'm having some issues with sourcemaps not being loaded correctly; will work on that more tonight after I make progress on another ticket.

**Comment by pl12133** [comment]
It will take a coordinated effort to merge this since it will be a large breaking change.

@billneff79 what do you think of publishing this as a `@beta`?

**Comment by billneff79** [comment]
I think beta is a good idea

**Comment by pl12133** [comment]
Published version `1.2.0-beta.1` tagged as `beta`.

If we need to publish any changes in between now and the beta becoming stable we can revert this PR and track beta on a `beta` branch.

**Pull request merged**